### ISSUE TITLE
Game does not come with compilation instructions

### ISSUE THREAD

**Issue by Wuzzy2** [CorsixTH/CorsixTH#915]
The game does not seem to be distributed together with compilation instructions.
Instead, the `readme.txt` file points to https://github.com/CorsixTH/CorsixTH/wiki/How-To-Compile.

This is bad practice IMO. Compilation instructions should be distributed along the source code. People are not always online.

Please don't forget that GitHub uses GitHub-flavored Markdown, and NOT just Markdown, so some cleanup might be needed before committing.

**Comment by Alberth289346** [comment]
Nobody ever checks files from the project data, so while we could move the compile instructions to a file and distribute them, they would be less easy to find, and get obsolete in no time.

I agree the wiki is not ideal, but it's easy to find, and keeping it more or less up-to-date is less effort.

**Comment by Wuzzy2** [comment]
Well, in the wiki you *could* simply a link to the readme or whatever file from the wiki.

**Comment by TheCycoONE** [comment]
Your scenario of a developer without the internet seems contrived - how did they get the source, and how are they going to get the dependencies that they are reading about in the build instructions? And if they know ahead of time to get all that stuff, why wouldn't they just save the wiki page?

**Comment by Wuzzy2** [comment]
Seriously, what is wrong of having all neccessary documentation readily available in the project itself? I don't see any harm done by that, only benefits.
I also don't see the point in having documentation *exclusively* on the wiki.

Why do you assume that developers are online 24/7? There can be many reasons why one can be offline: You are traveling with laptop and there are no WiFi access points, the Internet access at home broke down, you broke your configuration files, etc.

Also, compiling is not neccessarily a one-time process. Maybe someone already compiled it a long time ago, but now it is outdated. One got the source, then lost Internet connection and now wants to compile it. It would be annoying to wait for Internet again.

Etc. etc. etc. There are any possible scenarios where devs would be glad to have the docs available without Internet access. Saving the wiki page would be a smart move, but then I just point out another scenario in which one dev did not so. Maybe didn't know it is only available online or even just simply forgotten.
Long story short: It is more convenient and more fool-proof to have all neccessary documentation distributed with the game itself.

**Comment by TheCycoONE** [comment]
The issue is that it moves the maintenance burden from the community (anyone with an account can update the wiki) to the very small group of us that develop the game. If you feel passionately enough to produce a pr with some simple easy to maintain build instructions in the readme that would be ok. Feel free to copy appveyor.yml's script for Windows.

**Comment by Alberth289346** [comment]
> Long story short: It is more convenient and more fool-proof to have all neccessary documentation
> distributed with the game itself.

This is not necessarily true. If I have a version from 2013, with VS2013, and now I have VS2015, the readme file is definitely not updated to the new situation.
The wiki is more likely to be up-to-date wrt current build tools.

**Comment by lewri** [comment]
I believe that as this one has lapsed, it may be better to close this.
Or at least add a wontfix label.
